airbyte-lib/airbyte_lib/_processors.py

```diff
@@ -203,12 +206,18 @@ def process_airbyte_messages(
                 # Type.LOG, Type.TRACE, Type.CONTROL, etc.
                 pass
 
+        # Add empty streams to the dictionary, so we create a destiantion table for it
+        for stream_name in self._expected_streams:
+            if stream_name not in stream_batches.keys():
+                if DEBUG_MODE:
+                    print(f"Stream {stream_name} has no data")
+                stream_batches[stream_name] = []
+
         # We are at the end of the stream. Process whatever else is queued.
         for stream_name, stream_batch in stream_batches.items():
-            if stream_batch:
-                record_batch = pa.Table.from_pylist(stream_batch)
-                self._process_batch(stream_name, record_batch)
-                progress.log_batch_written(stream_name, len(stream_batch))
+            record_batch = pa.Table.from_pylist(stream_batch)
+            self._process_batch(stream_name, record_batch)
+            progress.log_batch_written(stream_name, len(stream_batch))
```

Comment:
This block makes sense to me. If we run into issues with the processor not being able to handle empty batches, we could evaluate other options - but this seems sound, well commented, and not overly complex.